industry_benchmarks/utils/data_gathering.py

```diff
@@ -444,14 +445,224 @@ def gather_ligand_scores(
+    """
+    # grab the gufe key of the chemical systems used in the inputs
+    unit_result = list(result["unit_results"].values())[0]
+    state_a_key = unit_result["inputs"]["stateA"][":gufe-key:"]
+    mapping_key = unit_result["inputs"]["ligandmapping"][":gufe-key:"]
+    # work out which system this is in the alchemical network
+    system_look_up = dict((str(node.key), node) for node in alchemical_network.nodes)
+    mapping_look_up = dict((str(edge.mapping.key), edge.mapping) for edge in alchemical_network.edges)
+    # build the transform
+    if any([isinstance(comp, gufe.ProteinComponent) for comp in system_look_up[state_a_key].components.values()]):
+        phase = "complex"
+    else:
+        phase = "solvent"
+
+    ligmap = mapping_look_up[mapping_key]
+
+    name = f"{phase}_{ligmap.componentA.name}_{ligmap.componentB.name}"
+    return name
+
+
+def  process_results(results_folders: list[pathlib.Path], output_dir: pathlib.Path, alchemical_network: gufe.AlchemicalNetwork) -> list[str]:
+    """
+    Loop over the results folders extracting the required information and moving it to the output folder.
+
+    Returns
+    -------
+        failed_results: list[str]
+        A list of edge names which have missing results
+    """
+    # workout the expected number of results
+    # assuming 3 repeats of each solvent and complex transformation
+    missing_results = []
+    all_results = {}
+    for edge in alchemical_network.edges:
+        for phase in ["complex", "solvent"]:
+            all_results[f"{phase}_{edge.stateA.components['ligand'].name}-{edge.stateB.components['ligand'].name}"] = []
+    expected_results = len(all_results) * 3
+
+    # map the transformation to the results files
+    for results_folder in results_folders:
+        for results_file in tqdm.tqdm(results_folder.glob("*.json"), desc=f"Processing results in {results_folder}", ncols=80):
+            # run checks on the end results
+            result = load_results_file(file_name=results_file)
+            if result is not None:
+                # find the cleaned up results file
+                simulation_data_file = find_data_folder(result=result)
+                # work out the name of the transform and add it to the dict
+                transformation_name = get_transform_name(result=result, alchemical_network=alchemical_network)
+                # collect the paths to the results files and the structural
+                if transformation_name in all_results and simulation_data_file is not None:
+                    all_results[transformation_name].append((results_file, simulation_data_file))
+                elif transformation_name not in all_results:
+                    print(f"Found a result for {transformation_name} which was not expected")
+                    continue
+    # make sure we found the expected number of results
+    assert sum([len(v) for v in all_results.values()]) == expected_results
```

Comment:
This will only be true if the entire network is finished, instead, report the total number of complete edges we have found results for and check if the final network built from the results is connected.

Reply:
Yes that would be great! If the network fixing script was used we would expect some failed edges (e.g. edges that are present in the AlchemicalNetwork but don't have any result .json files. So a check for connectivity instead would be great!

Reply:
Added the check in [55dd4f4](https://github.com/OpenFreeEnergy/IndustryBenchmarks2024/pull/179/commits/55dd4f41b5d01317c1b2221a00ac1ae98953b76c)